.ado/publish.yml

```diff
@@ -5,38 +5,156 @@
+
+  - script: |
       npm install -g @vscode/vsce
-    displayName: Install VSCE Pub Prereqs
-    
+    displayName: Install Prereqs for VSCode Extension
+    condition: eq(variables['Agent.OS'], 'Darwin')
+   
   - script: |
       python build.py --wasm --npm --vscode
     displayName: Build VSCode Extension
+    condition: eq(variables['Agent.OS'], 'Darwin')
+
+  - script: |
+      python build.py --pip
+    displayName: Build Platform-Dependent Py Packages
       
+  - script: |
+      python build.py --jupyterlab
+    displayName: Build Platform-Independent Py Packages
+    condition: eq(variables['Agent.OS'], 'Windows_NT')
+
   - script: |
       cp ../LICENSE.txt .
       vsce package $(Ver.Major).$(Ver.Minor).$(Ver.Patch)
     displayName: Pack VSCode Extension
     workingDirectory: '$(System.DefaultWorkingDirectory)/vscode'
+    condition: eq(variables['Agent.OS'], 'Darwin')
+
+  - script: |
+      ls vscode/*
+      ls vscode/out/*
+    displayName: List VSCode Extension
+    condition: eq(variables['Agent.OS'], 'Darwin')
+
+  - publish: $(System.DefaultWorkingDirectory)/vscode/qsharp-vscode-$(Ver.Major).$(Ver.Minor).$(Ver.Patch).vsix
+    artifact: VSIX
+    displayName: Upload VSCode Extension Artifact
+    condition: eq(variables['Agent.OS'], 'Darwin')
+
+  - publish: $(System.DefaultWorkingDirectory)/vscode/out/extension.js
+    artifact: VSIX.EntryPoint
+    displayName: Upload VSCode Ext EntryPoint Artifact
+    condition: eq(variables['Agent.OS'], 'Darwin')
+
+  - script: |
+      dir target\wheels\*
+    displayName: List Py Packages on Win
+    condition: eq(variables['Agent.OS'], 'Windows_NT')
+
+  - script: |
+      ls target/wheels/*
+    displayName: List Py Packages on non-Win
+    condition: ne(variables['Agent.OS'], 'Windows_NT')
+
+  - publish: $(System.DefaultWorkingDirectory)/target/wheels
+    artifact: Wheels.Mac
+    displayName: Upload Python Artifacts Mac
+    condition: eq(variables['Agent.OS'], 'Darwin')
+
+  - publish: $(System.DefaultWorkingDirectory)/target/wheels
+    artifact: Wheels.Win
+    displayName: Upload Python Artifacts Win
+    condition: eq(variables['Agent.OS'], 'Windows_NT')
+
+
+- job: "Publish"
+  dependsOn: Build
+  pool: 
+    vmImage: 'windows-latest'
+
+  steps:
+  - download: current
+    artifact: Wheels.Win
+    displayName: Download Python Artifacts Win
+
+  - download: current
+    artifact: Wheels.Mac
+    displayName: Download Python Artifacts Mac
+
+  - script: |
+      mkdir target\wheels
+      move ..\Wheels.Win\*.whl target\wheels
+      move ..\Wheels.Mac\*.whl target\wheels
+      dir target\wheels\*
+    displayName: Move Py Artifacts to Publishing Dir
+
+  - download: current
+    artifact: VSIX
+    displayName: Download VSCode Extension Artifact
+
+  - download: current
+    artifact: VSIX.EntryPoint
+    displayName: Download VSCode Ext EntryPoint Artifact
+
+  - script: |
+      md vscode\out
+      move ..\VSIX.EntryPoint\extension.js vscode\out
```

Comment:
Again, I don't understand the need for this. The extension javascript should already be in the VSIX package.

Reply:
Without this the publishing command fails like this:
```
 ERROR  Extension entrypoint(s) missing. Make sure these files exist and aren't ignored by '.vscodeignore':
  extension/out/extension.js
```

Reply:
Is the file present? This is what I see on mine (below). Maybe the last line in the .vscodeignore ('../**/*') is causing this when published outside the repo?

<img width="225" alt="image" src="https://github.com/microsoft/qsharp/assets/993909/eee6e311-e72d-44ef-9e67-1e3fe196cfc8">

Reply:
Resolved offline.